src/containers/FactoryLoader.tsx

```diff
@@ -143,6 +142,25 @@ export class FactoryLoaderContainer extends React.PureComponent<Props, State> {
     }
   }
 
+  private async openIde(): Promise<void> {
+    const { history, workspace } = this.props;
+    if (!workspace || WorkspaceStatus[workspace.status] !== WorkspaceStatus.RUNNING) {
+      return;
+    }
+    this.setState({ currentStep: LoadFactorySteps.OPEN_IDE });
+    try {
+      await this.props.requestWorkspace(workspace.id);
+    } catch (e) {
+      this.showAlert(`Getting workspace detail data failed. ${e}`);
+    }
+    await delay();
+    history.push(buildIdeLoaderPath(workspace));
+  }
+
+  private getCreatePolicy(attrs: { [key: string]: string }): CreatePolicy {
+    return attrs['policies.create'] as CreatePolicy || DEFAULT_CREATE_POLICY;
```

Comment:
Here you assume that `attrs['policies.create']` has a value that you expect. But it's not necessarily true, there may be any string value. I suggested you add a type guard function similar to described in [typescript handbook](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards)

Reply:
I have done it.
![Screenshot from 2021-02-22 16-49-47](https://user-images.githubusercontent.com/6310786/108725006-9abdaf80-752e-11eb-99ab-9f1837aaea9c.png)